backend/src/apiserver/server/pipeline_server_test.go

```diff
@@ -261,6 +263,48 @@ func TestListPipelineVersion_NoResourceKey(t *testing.T) {
 	assert.Equal(t, "Invalid input error: ResourceKey must be set in the input", err.Error())
 }
 
+func TestCreatePipelineVersionDontUpdateDefault(t *testing.T) {
+	viper.Set(common.UpdatePipelineVersionByDefault, "false")
+	defer viper.Set(common.UpdatePipelineVersionByDefault, "false")
+	httpServer := getMockServer(t)
+	// Close the server when test finishes
+	defer httpServer.Close()
+
+	clientManager := resource.NewFakeClientManagerOrFatal(
+		util.NewFakeTimeForEpoch())
+	resourceManager := resource.NewResourceManager(clientManager)
+
+	pipelineServer := PipelineServer{
+		resourceManager: resourceManager, httpClient: httpServer.Client(), options: &PipelineServerOptions{CollectMetrics: false}}
+	pipelineVersion, err := pipelineServer.CreatePipelineVersion(
+		context.Background(), &api.CreatePipelineVersionRequest{
+			Version: &api.PipelineVersion{
+				PackageUrl: &api.Url{
+					PipelineUrl: httpServer.URL + "/arguments-parameters.yaml"},
+				Name: "argument-parameters",
+				ResourceReferences: []*api.ResourceReference{
+					&api.ResourceReference{
+						Key: &api.ResourceKey{
+							Id:   "pipeline",
+							Type: api.ResourceType_PIPELINE,
+						},
+						Relationship: api.Relationship_OWNER,
+					}}}})
+
+	assert.Nil(t, err)
+	assert.NotNil(t, pipelineVersion)
+	assert.Equal(t, "argument-parameters", pipelineVersion.Name)
+	newPipelineVersion, err := resourceManager.GetPipelineVersion(
```

Comment:
It this really possible with the mockServer? 

replacing with 
``` _, err = resourceManager.GetPipeline(newPipelineVersion.PipelineId)```

result in ``` Error:      	Expected nil, but got: &util.UserError{internalError:(*errors.fundamental)(0xc000472200), externalMessage:"Pipeline pipeline not found.", externalStatusCode:0x5} ```

Not sure how the mock servers works, in this case.

Reply:
My guess is that the version API tests are sort of hacking, because they never create the pipeline resource. Not sure why it doesn't report a pipeline not found error.

I'd suggest avoid this hack by properly creating a pipeline first:
https://github.com/kubeflow/pipelines/blob/9da4a40ff12428dd13562a72db5fb22d4ccf019b/backend/src/apiserver/server/pipeline_server_test.go#L59

If this doesn't work, you can log a to-do as an issue.

Reply:
Thanks! Updated and fixed it.